joomla/joomla-cms
# [4.0] Fix multilang module

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/24339

### Summary of Changes
* Remove hardcoded reference to mod_multilangstatus in mod_status
* Remove no longer used helper class in mod_multilangstatus

Bringing the behavior back to a regular module.

### Testing Instructions (before PR)
First Scenario when language filter plugin is disabled:
* Try to enable the multilanguage status module and assign it to a position (eg "status")
* Notice the not so nice appearance of the icon
* Refresh page
* Module icon gone again

Second scenario when language filter plugin is enabled:
* If you did the first scenario, you now have two multilang icons
* Try to disable the multilanguage status module in the module manager
* First icon vanished, second one still visible
* Refresh page
* Both icons visible again. Module published again in the manager


### Expected result
Bevahior like any other module


### Actual result
Prior to PR: It seems to be living and doing its own thing 😨 

After PR: Regular behavior: 
* If module is assigned to a position and published -> appears.
* If module is not assigned to a position and/or not published -> doesn't appear.


### Documentation Changes Required
None
